sqlx-core/src/mysql/connection/executor.rs

```diff
@@ -18,15 +18,16 @@ use crate::mysql::protocol::statement::{
 };
 use crate::mysql::protocol::text::{ColumnDefinition, ColumnFlags, Query, TextRow};
 use crate::mysql::protocol::Packet;
+use crate::mysql::statement::MySqlStatement;
 use crate::mysql::{
     MySql, MySqlArguments, MySqlColumn, MySqlConnection, MySqlRow, MySqlTypeInfo, MySqlValueFormat,
 };
 use crate::statement::StatementInfo;
 
 impl MySqlConnection {
-    async fn prepare(&mut self, query: &str) -> Result<u32, Error> {
-        if let Some(statement) = self.cache_statement.get_mut(query) {
-            return Ok(*statement);
+    async fn prepare(&mut self, query: &str) -> Result<&mut MySqlStatement, Error> {
+        if self.cache_statement.contains_key(query) {
+            return Ok(self.cache_statement.get_mut(query).unwrap());
```

Comment:
I think this does the cache lookup twice instead of once previously, is there a particular reason?

Reply:
Borrow checker...

Reply:
So we can't do partial borrows yet. The `get_mut` returns `&mut T`. Even if we return the `&mut T` immediately, Rust thinks we still have a mutable reference to the cache in the end of the function, and doesn't let us to take more references...

Hopefully this will be fixed in future Rust versions.